duckduckgo/content-scope-scripts
# New C-S-S webcompat intervention: modifyLocalStorage

Adds new webcompat intervention to modify localStorage entries on specified domains. At the moment `delete` is the only action implemented, but extensibility is there to add other actions (`create`, `update`) if necessary in the future. This feature is only enabled on iOS & Android.

**Asana Task/Github Issue:** https://app.asana.com/0/246491496396031/1207836782368191/f
**Corresponding privacy config changes:** https://github.com/duckduckgo/privacy-configuration/pull/2283

Testing steps:
There are a couple ways to test this:
Option A (easiest): Run the integration tests included in this PR to check against the privacy test page also included in this PR
1.  Checkout this branch, run `npm i` then run `npm run test-int`. This will use puppeteer to check that the tests [here](https://github.com/duckduckgo/content-scope-scripts/pull/1026/files#diff-e53655b3e5ef75e4458ede138a9448187cec91f4f3217d98119de227e7a0295f) succeed.

Option B: 
1. Clone https://github.com/duckduckgo/privacy-test-pages and in package.json, point https://github.com/duckduckgo/privacy-test-pages/blob/main/package.json#L24 at this branch and `npm i`
2. Start the privacy test page local server by running `npm run start` from that directory
3. Create an iOS or Android dev build pointing C-S-S at this branch and privacy configuration at https://github.com/duckduckgo/privacy-configuration/pull/2283 and run it on the simulator
4. From the simulator, navigate to localhost:3000/content-scope-scripts/webcompat/pages/modify-localstorage.html and see that tests are green
5. Alternatively, from the simulator, navigate to any old [MSN article page](https://www.msn.com/en-us/sports/nfl/how-saquon-barkley-gave-eagles-coach-a-hard-lesson-in-accountability-while-saving-season/ar-AA1qPl89?ocid=hpmsn&cvid=81d6bc9eb97e442ba7de62c248189b23&ei=12) and refresh the page two or more times. See that the `Expand Article` button persists across reloads.
